src/libs/Navigation/AppNavigator/ModalStackNavigators.tsx

```diff
@@ -0,0 +1,293 @@
+import {ParamListBase} from '@react-navigation/routers';
+import {CardStyleInterpolators, createStackNavigator, StackNavigationOptions} from '@react-navigation/stack';
+import React, {useMemo} from 'react';
+import type {
+    AddPersonalBankAccountNavigatorParamList,
+    DetailsNavigatorParamList,
+    EditRequestNavigatorParamList,
+    EnablePaymentsNavigatorParamList,
+    FlagCommentNavigatorParamList,
+    MoneyRequestNavigatorParamList,
+    NewChatNavigatorParamList,
+    NewTaskNavigatorParamList,
+    ParticipantsNavigatorParamList,
+    PrivateNotesNavigatorParamList,
+    ProfileNavigatorParamList,
+    ReferralDetailsNavigatorParamList,
+    ReimbursementAccountNavigatorParamList,
+    ReportDetailsNavigatorParamList,
+    ReportSettingsNavigatorParamList,
+    ReportWelcomeMessageNavigatorParamList,
+    RoomInviteNavigatorParamList,
+    RoomMembersNavigatorParamList,
+    SearchNavigatorParamList,
+    SettingsNavigatorParamList,
+    SignInNavigatorParamList,
+    SplitDetailsNavigatorParamList,
+    TaskDetailsNavigatorParamList,
+    TeachersUniteNavigatorParamList,
+    WalletStatementNavigatorParamList,
+} from '@navigation/types';
+import useThemeStyles from '@styles/useThemeStyles';
+import SCREENS from '@src/SCREENS';
+
+type Screens = Record<string, () => React.ComponentType>;
```

Comment:
```suggestion
type Screens = Partial<Record<Screen, () => React.ComponentType>>;
```

I think we can improve this type to only accept the screens names, you add this in `src/SCREENS.ts`:

```ts
import DeepValueOf from './types/utils/DeepValueOf';

// ...

type Screen = DeepValueOf<typeof SCREENS>;

export default SCREENS;
export {PROTECTED_SCREENS};
export type {Screen};
```

Also change here:

```ts
        return (
            <ModalStackNavigator.Navigator screenOptions={defaultSubRouteOptions}>
                {Object.keys(screens as Required<Screens>).map((name) => (
                    <ModalStackNavigator.Screen
                        key={name}
                        name={name}
                        getComponent={(screens as Required<Screens>)[name as Screen]}
                    />
                ))}
            </ModalStackNavigator.Navigator>
        );
```

WDYT?